app/app.go

```diff
@@ -259,6 +262,15 @@ func New(
 	cdc := encodingConfig.Amino
 	interfaceRegistry := encodingConfig.InterfaceRegistry
 
+	// set the protocol version to a specific value if specified. This is used
+	// for testing purposes.
+	if pvo := appOpts.Get(SetProtocolVersionOptionKey); pvo != nil {
+		baseAppOptions = append(baseAppOptions, func(ba *baseapp.BaseApp) {
+			appVersion := pvo.(uint64)
+			ba.SetProtocolVersion(appVersion)
```

Comment:
this is done in v1.x in https://github.com/celestiaorg/celestia-app/blob/2dbfabf1849e166974c1287c35b43e5e07727643/cmd/celestia-appd/cmd/root.go#L249-L251

but not on main